packages/material-ui/src/Switch/Switch.test.js

```diff
@@ -87,4 +88,50 @@ describe('<Switch />', () => {
 
     expect(getByRole('checkbox')).to.have.property('checked', false);
   });
+
+  describe('with FormControl', () => {
+    describe('enabled', () => {
+      it('should not have the disabled class', () => {
+        const { getByTestId } = render(
+          <FormControl>
+            <Switch data-testid="root" />
+          </FormControl>,
+        );
+
+        expect(getByTestId('root')).not.to.have.class(classes.disabled);
+      });
+
+      it('should be overridden by props', () => {
+        const { getByTestId } = render(
+          <FormControl>
+            <Switch data-testid="root" disabled />
+          </FormControl>,
+        );
+
+        expect(getByTestId('root')).to.have.class(classes.disabled);
+      });
+    });
+
+    describe('disabled', () => {
+      it('should have the disabled class', () => {
+        const { getByTestId } = render(
+          <FormControl disabled>
+            <Switch data-testid="root" />
+          </FormControl>,
+        );
+
+        expect(getByTestId('root')).to.have.class(classes.disabled);
```

Comment:
What about running the assertion on the input?

Reply:
@eps1lon Proposes: `expect(getByRole('checkbox')).to.be.disabled`  or `expect(getByRole('checkbox')).to.have.property('disabled', true)` if the first matcher isn't implemented in chai-dom.

Reply:
> What about running the assertion on the `<input>` element rather than the class name?

Really good idea, thanks